scipy/optimize/cython_optimize/Newton/halley.c

```diff
@@ -0,0 +1,47 @@
+#include <math.h>
```

Comment:
No reason, in fact, I would be very happy to move these to `Zeros/` because I think it would make it cleaner. I didn't want to pollute what was already there, and I capitalized it because I was following the patter, since `Zeros/` is capitalized.